modules/configuration/ajax/process.php

```diff
@@ -27,39 +27,100 @@
 $client = new NDB_Client();
 $client->makeCommandLine();
 $client->initialize();
-
 $DB =& Database::singleton();
 foreach ($_POST as $key => $value) {
     if (is_numeric($key)) { //update
         if ($value == "") {
             $DB->delete('Config', array('ID' => $key));
         } else {
-            $DB->update(
-                'Config',
-                array('Value' => $value),
-                array('ID' => $key)
-            );
+            // if no duplicate value then do updating
+            if (duplicateExistsInDropdown($key, $value)) {
+                $DB->update(
+                    'Config',
+                    array('Value' => $value),
+                    array('ID' => $key)
+                );
+            } else {
+                   header("HTTP/1.1 400 Bad Request");
+                   exit();
+            }
         }
     } else { //add new or remove
         $keySplit   = explode("-", $key);
         $valueSplit = explode("-", $value);
         if ($keySplit[0] == 'add') {
             if ($value !== "") {
-                $DB->insert(
-                    'Config',
-                    array(
-                     'ConfigID' => $keySplit[1],
-                     'Value'    => $value,
-                    )
-                );
+                if (countDuplicate($keySplit[1], $value) == '0') {
+                    $DB->insert(
+                        'Config',
+                        array(
+                         'ConfigID' => $keySplit[1],
+                         'Value'    => $value,
+                        )
+                    );
+                } else {
+                        header("HTTP/1.1 303 Duplicate value");
+                        exit();
+                }
             }
         } elseif ($valueSplit[0] == 'remove') {
             $DB->delete('Config', array('ID' => $valueSplit[1]));
         }
     }
 }
+/**
+ * Check Duplicate value
+ *
+ * @param string $key   The value of the key
+ * @param string $value The value of the value
+ *
+ * @return int $result
+ */
+function countDuplicate($key,$value)
+{
+       $DB     =& Database::singleton();
+       $result = $DB->pselectOne(
+           "SELECT COUNT(*) FROM Config WHERE ConfigID =:ConfigID AND Value =:Value",
+           array(
+            ':ConfigID' => $key,
+            ':Value'    => $value,
+           )
+       );
+       return $result;
+}
+/**
+ * Check dropdown list Duplicate value
+ *
+ * @param string $id    The value of the id
+ * @param string $value The value of the value
+ *
+ * @return boolean return true if there is no Duplicate value
+ */
+function duplicateExistsInDropdown($id,$value)
```

Comment:
this is confusing. if the function returns true when duplicate does not exist, then the name of the function shouldn't be 'duplicateExists'